### DISCUSSION TITLE
Git Graph to see all commits across all branches

### DISCUSSION THREAD

**Discussion by isaiahdahl** [community/community#10515] [Pull Requests]
Just moved our team from bitbucket to Github. I've got to say the feature I am missing most is this commit's view.

![CleanShot 2022-01-24 at 17 27 38](https://user-images.githubusercontent.com/23046374/150893637-0753583a-3820-4341-981a-147ae43a5766.png)

I could look at the commits across all branches and see a Git Graph about where everything was at. 

Even if there isn't a git graph, I would find it very useful to have All Branches as an option here.

![CleanShot 2022-01-24 at 17 30 06](https://user-images.githubusercontent.com/23046374/150893842-5abb504f-4d23-4652-aada-2ab778175863.png)

**Comment by Katsute** [comment]
Maybe check out the `Insights > Network` tab on your repository

**Reply by isaiahdahl** [reply]
Not available on Private Repos. Also not really what I'm looking for. The graph is actually less important, having a place to list "All Commits" is more what I think is missing for me.

**Reply by nathabonfim59** [reply]
For anyone searching for this feature, you can access the "Activity" section. It's available even for private repositories. You'll find it in the sidebar, right after the "Readme" icon.

![image](https://github.com/community/community/assets/21281852/bc91f51b-acc9-4fc8-9552-3edb80df3a34)

**Reply by gnowland** [reply]
But there's no branch "subway map", or am I missing something?

**Reply by BinToss** [reply]
> But there's no branch "subway map", or am I missing something?

The Network graph is a bit better, but isn't as good as Le Git Graph, unless you want to compare forks.

**Reply by araujorayza** [reply]
> > But there's no branch "subway map", or am I missing something?
> 
> The Network graph is a bit better, but isn't as good as Le Git Graph, unless you want to compare forks.

I don't see a graph, It's basically a list. Do I need to do something else to show it?

**Comment by carun** [comment]
I just moved from Bitbucket to GitHub and I miss this dearly. We need the ability to view all the branches (as labels) and their commits as graph. Equivalent git command: `git log --graph --oneline --all`

**Comment by NirmalScaria** [comment]
You could use Le Git Graph, a browser extension that does exactly this.

[![Le git graph demo][1]][1]


Install the extension from here : [https://chrome.google.com/webstore/detail/le-git-graph-commits-grap/joggkdfebigddmaagckekihhfncdobff][2]

It will add a new "commits" section to every GitHub repo you open. Open the commits graph and there, all commits across branches will be listed along with the git graph.


[![Le git graph demo 2][3]][3]

Hope it helps! 

  [1]: https://i.stack.imgur.com/jLC3u.jpg
  [2]: https://chrome.google.com/webstore/detail/le-git-graph-commits-grap/joggkdfebigddmaagckekihhfncdobff
  [3]: https://i.stack.imgur.com/tesWi.jpg

**Reply by carun** [reply]
Hey @NirmalScaria this is fantastic. There appears to be some bugs. I'll email you the details.

**Reply by m0han-suresh** [reply]
How secure is this? Browser extensions can steal data if they want isit?

**Reply by NirmalScaria** [reply]
@m0han-suresh  Hey, I am the developer of this extension. I would like to clarify on that. Browser extensions are always open source. (Some might obfuscate, but there is really no way to properly compile anything). And I have made sure this extension is open source and not obfuscated at all. Anyone can see the source code and it does not collect any data.

One possible attack surface is the OAuth workflow. And technically, I admit, it is possible to steal data for the app owner if you use OAuth for login. But, if you are working as part of an organisation or anything serious at all, you have the option to use a PAT that you could generate manually. There will be no way for anyone to steal any data. The instructions are there in the github readme of the extension source code.

Thank you.

**Comment by vornde01** [comment]
The chrome extention looks great, but it's blocked at my company.

**Reply by 13steinj** [reply]
Similarly, doesn't run on enterprise instances, and uses firebase functions which means it phones home in some regard.

**Reply by NirmalScaria** [reply]
@13steinj  Hey, the firebase functions call is mainly for setting up OAuth. It was unavoidable to have some server running on the backend and being called from the extension for the authentication to work. (And the only other thing that calls the server is for usage statistics, and doesn't take any information.).
.
About the problem with enterprise instances, I am really not aware with the issue. It would be helpful if you could file it as an issue in the repo with details.

**Reply by 13steinj** [reply]
> About the problem with enterprise instances, I am really not aware with the issue. It would be helpful if you could file it as an issue in the repo with details.

It just quite literally doesn't load on any site that isn't github.com. It only connects, to github.com. Enterprise instances exist on separate domains. See refined-github for what they do to handle this (and authentication)-- they go the PAT route and support arbitrary domains.

Assuming I'm just missing something and it can load on enterprise instances-- going the PAT route, and having some way to disable statistics (with optional extension permissions of some sort) would go a long way. For enterprise instances, if the extension hits "anything except the domain of the site it's on", organizations would be hesitant to enable / allow this.

**Comment by MarkDaoust** [comment]
@NirmalScaria Awsome.

Any chance you can make it work for the "commits" tab on PRs too?

**Reply by NirmalScaria** [reply]
I didn't quite get it. Could you elaborate in a little more detail for what is expected? Perhaps file it as a new issue in the repo?

**Reply by MarkDaoust** [reply]
https://github.com/NirmalScaria/le-git-graph/issues/43

Thanks!

**Comment by brunomjm** [comment]
This feature is dearly missed in my company as well after migrating to GitHub Enterprise from Bitbucket...

**Comment by kizzie** [comment]
I miss this from bitbucket so much!

**Comment by yoshida-yuen-ijs** [comment]
I'd like this feature too. It's not very user-friendly that the graphs can't be viewed.

**Comment by andrea-masciotta-idsg** [comment]
Such a useful feature, missing it too after migration from BitBucket!
